### DISCUSSION TITLE
can't get it to work w/pan cam v1

### DISCUSSION THREAD

**Discussion by ewallis** [gtxaspec/wz_mini_hacks#525] [Q&A]
Using the `compile_image.sh pan` doesn't work as is. I had to set the download url manually in the file then it grabs it, creates demo.bin and all looks good. I copy the contents of SD_ROOT to a freshly fat32 formatted card and the demo.bin to the root of the card. Put card in camera, do reset, and setup through app. SSH server is not listening, nor is there any other indication that wz_mini_hacks is preset/working. What could I be doing wrong? Would love it get this working, thanks all.

**Comment by bjs-pdx** [comment]
Did you hear anything from the camera during first boot saying anything about wz_mini?  Did you configure SSH with your key?  Need more information if you would like help.  Did you do the firmware flash for the camera?  That's required for the Pan V1 and the regular V2.

**Reply by ewallis** [reply]
No, haven't heard anything audibly about wz_mini. Yes, I did the key but it doesn't get that far, gets a connection refused to port 22. I followed the directions about inserting the card and holding down the setup button when I plug it in. Am I missing something?

Here's the output of the `compile_image.sh pan`
```
wz_mini: this will download the latest firmware version from the vendor and compile a modified demo.bin.  Are you sure? [y/N]y
checking for tools
tools OK
create temporary work directory
check for local zip
local archive not found
downloading latest firmware
build for pan
--2023-05-14 14:33:23--  https://download.wyzecam.com/firmware/pan/demo_4.10.9.1851.bin.zip
Resolving download.wyzecam.com (download.wyzecam.com)... 172.67.129.82, 104.21.2.148, 2606:4700:3032::ac43:8152, ...
Connecting to download.wyzecam.com (download.wyzecam.com)|172.67.129.82|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 9154222 (8.7M) [application/zip]
Saving to: ‘./v2_ro/demo.zip’

./v2_ro/demo.zip                        100%[=============================================================================>]   8.73M  --.-KB/s    in 0.1s    

2023-05-14 14:33:24 (65.6 MB/s) - ‘./v2_ro/demo.zip’ saved [9154222/9154222]

extracting firmware to temporary work directory
Archive:  v2_ro/demo.zip
  inflating: ./v2_ro/demo_4.10.9.1851.bin  
  inflating: ./v2_ro/__MACOSX/._demo_4.10.9.1851.bin  
create firmware directory
unpack stock firmware image
+ ACTION=unpack
+ KERNEL_OFFSET=64
+ ROOTFS_OFFSET=2097216
+ DRIVER_OFFSET=5570624
+ APPFS_OFFSET=6225984
+ [ unpack = unpack ]
+ DEMO_IN=v2_ro/demo_4.10.9.1851.bin
+ OUT_DIR=v2_ro/fw_dir
+ dd if=v2_ro/demo_4.10.9.1851.bin of=v2_ro/fw_dir/kernel.bin skip=64 count=2097152 bs=1
2097152+0 records in
2097152+0 records out
2097152 bytes (2.1 MB, 2.0 MiB) copied, 3.7288 s, 562 kB/s
+ md5sum v2_ro/fw_dir/kernel.bin
bee7951307dc53edf59599cd15294a88  v2_ro/fw_dir/kernel.bin
+ dd if=v2_ro/demo_4.10.9.1851.bin of=v2_ro/fw_dir/rootfs.bin skip=2097216 count=3473408 bs=1
3473408+0 records in
3473408+0 records out
3473408 bytes (3.5 MB, 3.3 MiB) copied, 6.34733 s, 547 kB/s
+ md5sum v2_ro/fw_dir/rootfs.bin
59a04f8f9d4ff53e0a2edac74a47ac37  v2_ro/fw_dir/rootfs.bin
+ dd if=v2_ro/demo_4.10.9.1851.bin of=v2_ro/fw_dir/driver.bin skip=5570624 count=655360 bs=1
655360+0 records in
655360+0 records out
655360 bytes (655 kB, 640 KiB) copied, 1.19991 s, 546 kB/s
+ md5sum v2_ro/fw_dir/driver.bin
639f36987db72099380776aa21f64944  v2_ro/fw_dir/driver.bin
+ uname -s
+ [ Linux = Darwin ]
+ stat -c %s v2_ro/demo_4.10.9.1851.bin
+ IMAGE_END=11075648
+ dd if=v2_ro/demo_4.10.9.1851.bin of=v2_ro/fw_dir/appfs.bin skip=6225984 count=4849664 bs=1
4849664+0 records in
4849664+0 records out
4849664 bytes (4.8 MB, 4.6 MiB) copied, 8.87345 s, 547 kB/s
+ md5sum v2_ro/fw_dir/appfs.bin
fbdff46e659e5fa54af0cbae7f96623b  v2_ro/fw_dir/appfs.bin
replace factory kernel
pack firmware image with new kernel
+ ACTION=pack
+ KERNEL_OFFSET=64
+ ROOTFS_OFFSET=2097216
+ DRIVER_OFFSET=5570624
+ APPFS_OFFSET=6225984
+ [ pack = unpack ]
+ [ pack = pack ]
+ TMP_DIR=v2_ro/fw_dir
+ DEMO_OUT=demo.bin
+ dd if=/dev/zero of=v2_ro/fw_dir/kernel.bin bs=1 count=1 seek=2097151
1+0 records in
1+0 records out
1 byte copied, 0.000190785 s, 5.2 kB/s
+ cat v2_ro/fw_dir/kernel.bin v2_ro/fw_dir/rootfs.bin v2_ro/fw_dir/driver.bin v2_ro/fw_dir/appfs.bin
+ mkimage -A MIPS -O linux -T firmware -C none -a 0 -e 0 -n jz_fw -d v2_ro/fw_dir/flash.bin demo.bin
Image Name:   jz_fw
Created:      Sun May 14 14:33:44 2023
Image Type:   MIPS Linux Firmware (uncompressed)
Data Size:    11075584 Bytes = 10816.00 KiB = 10.56 MiB
Load Address: 00000000
Entry Point:  00000000
remove temporary work directory
demo.bin ready.  Please copy demo.bin to your memory card
```

**Reply by bjs-pdx** [reply]
"inserting the card and holding down the setup button when I plug it in. "

If everything worked (i.e. firmware update completed) you would hear a female voice saying something like wz_mini first boot extracting ...

My suggestion...factory reset the camera. Insert a freshly formatted sd card (32gb or less) containing the demo.bin and wz_mini files and try re-doing the firmware installation.  Then carefully listen as the camera reboots for the installation message.

In case you didn't see the post, the Pan Cams are limited in functionality...vertical movement (tilt) doesn't seem to work correctly.

You probably need a different version of the Wyze firmware too:

`Due to the security improvement on this firmware, you will need to go through the setup process for your camera if you go back to firmware 4.9.8.1002 or below. Because of this, we are removing those firmware files from this page.`

Link to firmware:  https://download.wyzecam.com/firmware/pan/demo_4.10.8.1002.zip  Note:  this is the version the compile script is looking for.

**Reply by ewallis** [reply]
I replaced link w/the one above and ran the compile_image.sh pan. I then copied contents of SD_ROOT and demo.bin to the root of a different freshly formatted 16GB card. Put card in camera and held setup button as I plugged it in, held for a second, then released tehe button. No dice, not sure what I'm doing wrong.

**Reply by bjs-pdx** [reply]
First, does it appear that you got the correct firmware file for the download?  Second, did you do a factory reset on the camera BEFORE trying to run the stuff on the SD card for wz_mini?  This is a required step.  Third, did you hold the reset button until the light on the camera changed to purple (at least I think that's the color for the Pan)?  One second isn't enough, the light has to change.

**Reply by ewallis** [reply]
I will try those things. At what point do you add it to the wyze app so it can get get on wifi?

**Reply by bjs-pdx** [reply]
You need a downgraded firmware first.  You should use the one you are downloading for wz_mini. But you need to do the factory reset before you can add the camera to wifi.  Then and only then would you install the wz_mini firmware.

**Reply by ewallis** [reply]
I can't seem to get it to downgrade. I followed wyze's instructions [here](https://support.wyze.com/hc/en-us/articles/360031490871-How-to-flash-your-Wyze-Cam-firmware-manually) and it just won't work.

**Reply by bjs-pdx** [reply]
Did you do the factory reset first?  Did you extract the firmware file from the zip?  Did you name it demo.bin and not something like demo.bin.bin?  Do you get the solid blue light when holding setup during the power up?  You need to give more information.  Would you also please confirm which camera?  Pan V1 or Pan V2?  Not Pan V3 right?

**Reply by ewallis** [reply]
First, I really appreciate your help. I'm trying to get it downgraded to 4.10.8.1002 (it's currently 4.10.9.1851) first by following Wyze's procedures:

format 2GB micro sd to fat32
download 4.10.8.1002 zip
extract zip and rename resulting bin to demo.bin
copy demo.bin to root of sd card
unplug camera, insert sd card. power camera while holding setup button, continue to hold until blue and yellow light are both lit (3-6 seconds)
let it sit 10 mins while hopefully downgrading version
add to app again, still 4.10.9.1851

I also attempted the above after factory resetting by holding setup button while on for 20 seconds.

**Reply by bjs-pdx** [reply]
Have you tried a different SD card?  Sounds like the camera isn't reading the file.

**Reply by ewallis** [reply]
Yes, I've tried the above w/a 16GB card. I assume it's reading the card ok because it records events on it.

**Reply by bjs-pdx** [reply]
Then it has to be the file.  What is the OS of your computer?  If Windows, have you turned on the option to show all file extensions?  Window typically hides them so you end up with a file called demo.bin.bin.

**Reply by ewallis** [reply]
Ubuntu 20.04, I've tried the above on windows as well

**Reply by bjs-pdx** [reply]
Are you sure it's a Pan V1?  Not V2?  Does it have a USB A port on the back?

The only other thing is that you may not be holding down the setup button completely.  Try using some small tool to press and hold the button while you power it on.

**Reply by ewallis** [reply]
Pan v1, yes, it has a usb A port on the back. I can feel the positive click when I hold the button down.

**Reply by bjs-pdx** [reply]
You may still want to try some tool to hold it down.  If it isn't held completely during the startup the firmware flash will fail. Make sure that the light goes blue.

**Reply by ewallis** [reply]
I tried, thanks for trying w/me, giving up on it for now.
